src/plots/Volume/avtDefaultRenderer.C

```diff
@@ -144,128 +154,157 @@ void
 avtDefaultRenderer::Render(
     const avtVolumeRendererImplementation::RenderProperties &props,
     const avtVolumeRendererImplementation::VolumeData &volume)
-{ 
+{
     const char *mName = "avtDefaultRenderer::Render: ";
 
-    if (VTKRen == NULL)
+    if (VTKRen == nullptr)
     {
-        debug1 << mName << "Default Renderer: VTKRen is NULL!";
+        debug1 << mName << "Default Renderer: VTKRen is nullptr!";
         EXCEPTION0(ImproperUseException);
     }
 
-    // 
-    // 2D data has no volume, so we don't try to render this. 
-    // 
+    // 2D data has no volume, so don't try to render.
     if (props.dataIs2D)
     {
-        debug5 << mName << "Cannot perform volume rendering on " 
+        debug5 << mName << "Cannot perform volume rendering on "
             << "2D data... returning";
         return;
     }
 
-    if (imageToRender == NULL)
+    if (imageToRender == nullptr)
     {
-        debug5 << mName << "Converting from rectilinear grid " 
+        debug5 << mName << "Converting from rectilinear grid "
             << "to image data";
 
-        //
-        // Our mapper requires a vtkImageData as input. We must 
-        // create one. 
-        //
-        int dims[3], extent[6];
-        ((vtkRectilinearGrid *)volume.grid)->GetDimensions(dims);
-        ((vtkRectilinearGrid *)volume.grid)->GetExtent(extent);
+        vtkRectilinearGrid* rgrid = vtkRectilinearGrid::SafeDownCast( volume.grid );
 
-        //
-        // We might be using a different scalar for opacity than 
-        // for color. We need to get both arrays. 
-        //
-        vtkDataArray *dataArr = volume.data.data;
-        vtkDataArray *opacArr = volume.opacity.data;
+        if( rgrid->GetDataObjectType() != VTK_RECTILINEAR_GRID )
+        {
+            EXCEPTION1(ImproperUseException,
+                       "Only vtkRectilinearGrid may be rendered.");
+        }
 
-        vtkRectilinearGrid *rgrid = (vtkRectilinearGrid *)volume.grid;
-        double spacingX = rgrid->GetXCoordinates()->GetTuple1(1)-
-            rgrid->GetXCoordinates()->GetTuple1(0);
-        double spacingY = rgrid->GetYCoordinates()->GetTuple1(1)-
-            rgrid->GetYCoordinates()->GetTuple1(0);
-        double spacingZ = rgrid->GetZCoordinates()->GetTuple1(1)-
-            rgrid->GetZCoordinates()->GetTuple1(0);
+        // Check for an implied transform.
+        // avtDataAttributes &inatts = GetInput()->GetInfo().GetAttributes();
```

Comment:
If the commented out section is reserved for future use, please add comments. Otherwise remove.

Reply:
Added comment - this class will be deprecated at some point.